pkg/controller/bastion/options.go

```diff
@@ -163,30 +162,95 @@ func determineImageID(shoot *gardencorev1beta1.Shoot, providerConfig *awsv1alpha
 	return "", fmt.Errorf("found no suitable AMI for machines in region %q", shoot.Spec.Region)
 }
 
-func determineInstanceType(ctx context.Context, awsClient *awsclient.Client) (string, error) {
-	offerings, err := awsClient.EC2.DescribeInstanceTypeOfferingsWithContext(ctx, &ec2.DescribeInstanceTypeOfferingsInput{})
+func determineInstanceType(ctx context.Context, imageID string, awsClient *awsclient.Client) (string, error) {
+	var preferredType string
+	imageInfo, err := getImages(ctx, imageID, awsClient)
 	if err != nil {
-		return "", fmt.Errorf("failed to list instance types: %w", err)
+		return "", err
 	}
 
-	types := make([]string, len(offerings.InstanceTypeOfferings))
-	for i, offering := range offerings.InstanceTypeOfferings {
-		types[i] = *offering.InstanceType
+	imageArchitecture := imageInfo.Architecture
+
+	// default instance type
+	switch *imageArchitecture {
+	case "x86_64":
+		preferredType = "t2.nano"
+	case "arm64":
+		preferredType = "t4g.nano"
+	default:
+		return "", fmt.Errorf("image architecture not supported yet")
 	}
 
-	// prefer t2.nano
-	for _, t := range types {
-		if t == "t2.nano" {
-			return t, nil
-		}
+	exist, err := checkInstanceTypeOffering(ctx, preferredType, awsClient)
+	if err != nil {
+		return "", err
+	}
+
+	if len(exist.InstanceTypeOfferings) != 0 {
+		return preferredType, nil
+	}
+
+	// filter t type instance
+	tTypes, err := checkInstanceTypeOffering(ctx, "t*", awsClient)
+	if err != nil {
+		return "", err
 	}
 
-	// fallback to the first (hopefully smallest) other general purpose instance type
-	for _, t := range types {
-		if strings.HasPrefix(t, "t") {
-			return t, nil
+	if len(tTypes.InstanceTypeOfferings) == 0 {
+		return "", fmt.Errorf("no t.* instance type available")
```

Comment:
```suggestion
		return "", fmt.Errorf("no t* instance type offerings available")
```
this also to make the error distinguishable from

Reply:
committed a6b392709b95eb54d9638ad54de25af7997148ce